addons/mrp/models/mrp_production.py

```diff
@@ -1451,15 +1451,17 @@ def button_mark_done(self):
         if self.env.context.get('mo_ids_to_backorder'):
             productions_to_backorder = self.browse(self.env.context['mo_ids_to_backorder'])
             productions_not_to_backorder = self - productions_to_backorder
+            close_mo = False
         else:
             productions_not_to_backorder = self
             productions_to_backorder = self.env['mrp.production']
+            close_mo = True
 
         self.workorder_ids.button_finish()
 
+        backorders = productions_to_backorder._generate_backorder_productions(close_mo=close_mo)
```

Comment:
Additional products are passed to the backorder. I think it should not :)

Reply:
@Whenrow Do you mean that if a user inserts an additional product in the original order the same product will appear in the backorder as well? If that's the case, there is actually a condition to exclude additional products in `_generate_backorder_production` (line 1397). I also tested it and works fine.

Reply:
@npi agreed but it's restrict to the `close_mo = False`
@mathieu-mgm Could you please give your preference here ? Should we pass additional product to backorders ?

Reply:
@Whenrow Ideally the additional components should be 
- in the backorder if they are added while the MO status is 'draft+confirmed' 
- not in the backorder if they are added while the MO status is in 'in progress+to close' 
@npi-odoo

Reply:
@mathieu-mgm There is already a strong separation of move between draft and ['confirmed'+'in progress+to close']
Couldn't we keep this behavior instead?

Reply:
@Whenrow Yes, but `close_mo` is set to `False` when creating the backorder (except if I missed some other ways of creating a backorder without going through `button_mark_done` in the simplified view)
@mathieu-mgm The current code passes the additional product in the backorder if it was created in the 'draft' state but not in the 'confirmed' one.

Reply:
@npi-odoo please correct the fix so as to pass additional components to the backorder. If they are not supposed to the user simply needs to set the to consume quantity to 0 
cc @amoyaux @Whenrow